src/comm/LinkManager.cc

```diff
@@ -504,8 +499,15 @@ void LinkManager::_updateConfigurationList(void)
 #endif
         // Save port name
         currentPorts << portInfo.systemLocation();
-        // Is this a PX4 and NOT in bootloader mode?
-        if (portInfo.vendorIdentifier() == SerialPortIds::px4VendorId && !portInfo.description().contains("BL")) {
+
+        QGCSerialPortInfo::BoardType_t boardType = portInfo.boardType();
+
+        if (boardType != QGCSerialPortInfo::BoardTypeUnknown) {
+            if (portInfo.isBootloader()) {
```

Comment:
I could probably make boardType return boot loader status to remove the double call. Let me see.